### ISSUE TITLE
tex_relative_path_to_data makes slashes in Windows style, which is not accepted by LaTeX.

### ISSUE THREAD

**Issue by IsabellLehmann** [nschloe/tikzplotlib#509]
Hi,

When I give an argument to `tex_relative_path_to_data`, the slashes in the .tex-file are in Windows style but LaTeX does not accept them. Would be good if they are these '/' also for Windows PCs.

Furthermore, I have no idea why there is `images/images` in the path. I think it should be there only once.

Here you find:
  * the Python code (make sure a folder `images` exists in the folder where you run the code),
```
import numpy as np
import matplotlib.pyplot as plt
import tikzplotlib

plt.imshow(np.random.randn(20,20))
tikzplotlib.save('images/test.tex', tex_relative_path_to_data='images', override_externals=True)
```

  * the output TeX (Pgfplots), and
```
% This file was created by tikzplotlib v0.9.8.
\begin{tikzpicture}

\begin{axis}[
tick align=outside,
tick pos=left,
x grid style={white!69.0196078431373!black},
xmin=-0.5, xmax=19.5,
xtick style={color=black},
y dir=reverse,
y grid style={white!69.0196078431373!black},
ymin=-0.5, ymax=19.5,
ytick style={color=black}
]
\addplot graphics [includegraphics cmd=\pgfimage,xmin=-0.5, xmax=19.5, ymin=19.5, ymax=-0.5] {images\images\test-000.png};
\end{axis}

\end{tikzpicture}
```

  * the expected TeX (Pgfplots).
```
% This file was created by tikzplotlib v0.9.8.
\begin{tikzpicture}

\begin{axis}[
tick align=outside,
tick pos=left,
x grid style={white!69.0196078431373!black},
xmin=-0.5, xmax=19.5,
xtick style={color=black},
y dir=reverse,
y grid style={white!69.0196078431373!black},
ymin=-0.5, ymax=19.5,
ytick style={color=black}
]
% the next line has the difference
\addplot graphics [includegraphics cmd=\pgfimage,xmin=-0.5, xmax=19.5, ymin=19.5, ymax=-0.5] {images/test-000.png};
\end{axis}

\end{tikzpicture}
```

Thanks for any suggestions besides manually adjusting the path in each image!
Isabell

**Comment by Blubbaa** [comment]
I am also not entirely sure of how this is supposed to behave. It seems like the option `tex_relative_path_to_data` will be just prepended to all paths used in the tikz file. So my workaround on Windows is to use `tex_relative_path_to_data` for the `./plots/` directory within my tex directory and another directory  `./plots/data/`  within, containing all additional data, images, tables etc. 
```Python3
import numpy as np
import matplotlib.pyplot as plt
import tikzplotlib

plt.imshow(np.random.randn(20,20))
tikzplotlib.save('./data/test.tex', tex_relative_path_to_data='plots', override_externals=True)
```
Doing so, will save all files to a sub directory called `./data/` and you will have to manually move the files to the corresponding tex directories (`./plots/` for tex file and `./plots/data/` for all others). This is still cumbersome but I'd rather move some files manually / via scripts then edit all auto-generated tikz files.

**Comment by IsabellLehmann** [comment]
Thanks @Blubbaa ! Will try it out next time I need to generate tikz images :)

Strange that this error does not occur with you ... You code generates for me:

**Comment by Blubbaa** [comment]
Ok, I did some investigations, it seems I encountered this issue before and just couldn't remember 😄 

There was an open PR (https://github.com/nschloe/tikzplotlib/pull/473) some time ago, which proposed to fix the problem here:
https://github.com/nschloe/tikzplotlib/blob/15f0bfce500714a956afafbebe19f6aa1669138d/src/tikzplotlib/_files.py#L38-L43
by adding some string replacement for Windows paths. Back then I have manually edited the local source from tikzplotlib and added this line in L42:
```Python3
    # FIX: for windows relative paths (see https://github.com/nschloe/tikzplotlib/pull/473)
    rel_filepath = str(rel_filepath).replace('\\', '/')
```
It seems like that this should have been fixed already, but apparently is not.

**Comment by nschloe** [comment]
I'll look at it later.

**Comment by Blubbaa** [comment]
I believe the support for .dat files was forgotten in the fix. Instead of `rel_filepath` here
https://github.com/nschloe/tikzplotlib/blob/1e30229a5ee6db8284718721c7c3ca8128d76778/src/tikzplotlib/_line2d.py#L291-L295
 `posix_filepath = rel_filepath.as_posix() ` should also be used instead.

Don't know if I should open a new issue or just reuse the old one.

**Comment by nschloe** [comment]
@Blubbaa Thanks for the notification. Fixed in https://github.com/nschloe/tikzplotlib/pull/533.
